### ISSUE TITLE
you couldn't translate Chinese into id property?

### ISSUE THREAD

**Issue by JimmyVV** [valeriangalliat/markdown-it-anchor#17]
what a shame~ author, why not you translate Chinese?

**Comment by valeriangalliat** [comment]
Duplicate of #10; the problem is the ID property is determined with strings.js, which seems to support only western languages.

What slugification module would you advise instead? I might switch from strings.js to something that have better language support it in a next major release (will be a breaking change).

Meanwhile, you can configure your own `slugify` function (takes the title text as argument, returns the ID), to make it support Chinese.

**Comment by JimmyVV** [comment]
en. maybe you misunderstand my mean. I just want you just to use text and mix with some hash to make sure the id is unique. not to translate Chinese. and I just do as you suggest

**Comment by valeriangalliat** [comment]
Makes sense, well in this case your `slugify` function can just be `title => title`, there's already code that adds `-1`, `-2` at the end of the ID in case an ID was already used.

I might default to this in the next major version, simple and easy. 👍

**Comment by airwin** [comment]
maybe a full solution:
```
var transliteration = require('transliteration')
var markdownItAnchorSlugCache = {}
// and use custom slugify function in markdown-it-anchor config
slugify: function(s) {
  var slug = transliteration.slugify(s, {lowercase: true, separator: '-', ignore: []})
  if (markdownItAnchorSlugCache[slug]) {
    slug += '-' + (markdownItAnchorSlugCache[slug]++)
  } else {
    markdownItAnchorSlugCache[slug] = 1
  }

  return slug
}
```

**Comment by nagaozen** [comment]
Fixed by version `5.0.0`.
